packages/hint-compat-api/src/compat-api-css-base.ts

```diff
@@ -0,0 +1,113 @@
+import { BrowserSupportCollection, FeatureInfo, BrowsersInfo } from './types';
+import { SimpleSupportStatement, VersionValue } from './types-mdn.temp';
+
+import meta from './meta/compat-api-css';
+import { browserVersions } from './helpers/normalize-version';
+import { CSSFeatureStatus } from './enums';
+
+/*
+ * ------------------------------------------------------------------------------
+ * Public
+ * ------------------------------------------------------------------------------
+ */
+
+export default abstract class BaseCompatApiCSS implements IHint {
+    public static readonly meta = meta;
+
+    private mdnBrowsersCollection: BrowserSupportCollection;
+    private compatApi: CompatApi;
+    private compatCSS: CompatCSS;
+
+    abstract getFeatureVersionValueToAnalyze(browserFeatureSupported: SimpleSupportStatement): VersionValue;
+    abstract isSupportedVersion(currentVersion: number, version: number): boolean;
+    abstract isVersionValueSupported(version: VersionValue): boolean;
+    abstract isVersionValueTestable(version: VersionValue): boolean;
+    abstract getStatusNameValue(): CSSFeatureStatus;
+
+    public constructor(context: HintContext<StyleEvents>, isCheckingNotBroadlySupported: boolean) {
+        this.mdnBrowsersCollection = userBrowsers.convert(context.targetedBrowsers);
+        this.compatApi = new CompatApi('css', this.mdnBrowsersCollection, isCheckingNotBroadlySupported);
+        this.compatCSS = new CompatCSS(context, this.testFeatureIsSupportedInBrowser.bind(this));
+
+        context.on('parse::end::css', async (styleParse: StyleParse) => {
+            await this.onParseCSS(styleParse);
+        });
+    }
+
+    private async onParseCSS(styleParse: StyleParse): Promise<void> {
+        const { resource } = styleParse;
+
+        this.compatCSS.setResource(resource);
+        await this.compatCSS.searchCSSFeatures(this.compatApi.compatDataApi, this.mdnBrowsersCollection, styleParse);
+    }
+
+    private async testFeatureIsSupportedInBrowser(browser: BrowsersInfo, feature: FeatureInfo): Promise<void> {
+        if (!this.compatApi.isBrowserToSupportPartOfBrowsersCollection(browser.browsersToSupport, browser.browserToSupportName)) {
+            return;
+        }
+
+        const browserFeatureSupported = this.compatApi.getSupportStatementFromInfo(browser.browserInfo, feature.prefix);
+
+        if (browserFeatureSupported) {
+            this.testVersionByBrowsers(browser, browserFeatureSupported, feature);
+        } else {
+            const message = `${feature.featureName} of CSS was never supported on any of your browsers to support.`;
+
+            await this.compatCSS.reportIfThereIsNoInformationAboutCompatibility(message, browser.browsersToSupport, browser.browserToSupportName, feature.featureName, feature.location);
+        }
+    }
+
+    private async testVersionByBrowsers(browser: BrowsersInfo, browserFeatureSupported: SimpleSupportStatement, feature: FeatureInfo) {
```

Comment:
I think we should keep always as much the order as possible:
1st. Browser Info
2nd. Feature Info
3rd other Info

What do you think?

Reply:
+1